src/components/menu/examples/menu-open-sub-menu-programatically.tsx

```diff
@@ -0,0 +1,113 @@
+
+/**
+ * Opening sub-menus programatically
+ *
+ * It is possible to open any sub-menu in the menu-hierarchy.
+ * This is done by using the parentItem property of the MenuItem class.
+ * @link item-constants.ts
+ */
+@Component({
+    tag: 'limel-example-menu-open-sub-menu-programatically',
+    shadow: true,
+    styleUrl: 'menu-open-sub-menu-programatically.scss',
+})
+export class MenuOpenSubMenuProgramaticallyExample {
+    private readonly rootItems: Array<MenuItem | ListSeparator> =
+        CascadingMenuItems;
+
+    @State()
+    private items: Array<MenuItem | ListSeparator> = this.rootItems;
+
+    @State()
+    private lastSelectedItem: string;
+
+    @State()
+    private selectedMenuItem: MenuItem;
+
+    @State()
+    private openMenu: boolean = false;
+
+    public render() {
+        return (
+            <Host>
+                {this.renderMenu()}
+                <limel-example-value
+                    label="Last selected item"
+                    value={this.lastSelectedItem}
+                />
+            </Host>
+        );
+    }
+
+    private renderMenu() {
+        return (
+            <div class="menu-container">
+                <limel-menu
+                    items={this.items}
+                    open={this.openMenu}
+                    selectedMenuItem={this.selectedMenuItem}
+                    onSelect={this.handleSelect}
+                    onNavigateMenu={async (
+                        event: LimelMenuCustomEvent<MenuItem>
+                    ) => {
+                        if (!event.detail) {
+                            this.items = this.rootItems;
+                        }
+                    }}
+                    onCancel={() => {
+                        this.items = this.rootItems;
+                        this.openMenu = false;
+                    }}
+                >
+                    <limel-button label="Menu" slot="trigger" />
+                </limel-menu>
+                <limel-button
+                    label="Shortcut to Lists"
+                    primary={true}
+                    onClick={this.buttonClick}
+                />
+            </div>
+        );
+    }
+
+    private buttonClick = () => {
+        const selectedItem: MenuItem = {
+            text: 'Lists',
+            parentItem: this.rootItems[0] as MenuItem,
+        };
+        const selectedItems: MenuItem[] = [
+            {
+                text: 'Numbered list',
+                icon: 'numbered_list',
+                parentItem: selectedItem,
+            },
+            {
+                text: 'Bullet list',
+                icon: 'bulleted_list',
+                parentItem: selectedItem,
+            },
+            {
+                text: 'Checklist',
+                icon: 'todo_list',
+                parentItem: selectedItem,
+            },
+        ];
+        this.selectedMenuItem = selectedItem;
+        this.items = selectedItems;
+        this.openMenu = true;
```

Comment:
I find it confusing that we have both `selectedItem` and `selectedItems` here.

The `selectedItems` array is, unless I'm mistaken, simply the items the menu should be showing at the moment. No one has selected them yet.

The name `selectedMenuItem` for the property that determines what sub-menu to show is also really confusing. Sure, I can see that you probably have selected that item to get to the state where the sub-menu is shown (unless the menu was opened programmatically), but I think a name like `submenuToDisplay` or `currentSubmenu` would be better. I know it's a little weird since that item doesn't really _contain_ the menu items shown, but that's what we convey to the user, so I think it's easier to grasp the code if the code is somewhat aligned with that concept.

Reply:
Makes sense, will fix that

Reply:
Hmm… hang on… in the `CascadingMenuItems` imported from `item-constants.ts`, the sub-menu items actually do _contain_ the items included in that sub-menu. Why is it different here?

Another difference is that all the items in `selectedItems` have a reference to their `parentItem`, which apparently isn't necessary when not opening the menu programmatically?

The "vanilla" example for cascading menu is so clean, you just supply a nested structure of items, and everything is just taken care of, but here, there is so much extra stuff to do, you have to create new items yourself, and replace the normal structure of items with this specially crafted one, and then remember to restore the normal structure again at the right moments (plural!).

Have you explored the possibility of keeping the simple "rootItems" and just telling the menu where to navigate to, somehow? I imagine you could provide it with a "path". I guess since it's all arrays, and `value` is optional, rather than a nice `'format.lists'`, you'd give it something like `'0.2'` or `'[0][2]'`, but still, it would be much less work on the consumer.

By the way, I just noticed this little curiosity, which I think highlights why it would be nice to not have the consumer do all this "manual labour":

https://github.com/Lundalogik/lime-elements/assets/154725/e698622a-73bf-4d0e-abed-f3ca118aab6d

Since I feel this feedback might be a little hard to hear, considering it might require some rework, I wan't to express that the cascading menu feature in itself is absolutely fantastic, and you've really done some great work here!

Is the ability to programmatically open the menu to any sub-menu critical for your use-case? If not, I would suggest simply leaving that functionality out for now, and add the feature once it's ready and stable.